### PR TITLE
Split pkg_zip implementation out of pkg.bzl

### PR DESCRIPTION
**Pull request by aiuto**
Another refactoring in the model of pkg_deb.

### CODE REVIEW

File: pkg/private/BUILD [hunk 1]
```diff
@@ -57,7 +57,7 @@ py_library(
     ],
     srcs_version = "PY3",
     visibility = [
-        "//pkg:__pkg__",
+        "//:__subpackages__",
```
Comment (nacl) [inline pkg/private/BUILD:60]: Could this just be `//pkg:__subpackages__` instead?

I'm not opposed to this change as-is, though.

File: pkg/private/BUILD [hunk 2]
```diff
@@ -70,7 +70,7 @@ py_library(
...
```
Comment (nacl) [inline pkg/private/BUILD:73]: Ditto.

**Review by nacl — APPROVED** [review]
LGTM other than a nitpick.

There are some other zipfiles being specified inside `tests/BUILD`.  Is the intent that package-specific tests go into `tests/$PACKAGE_NAME`, and more general ones remain in `tests/` directly?

File: pkg/private/BUILD (hunk 1 above)
Comment (aiuto) [reply pkg/private/BUILD:60]: I think the second one is too broad. Unless we want the internals visible to things like doc geneation.

**Comment by aiuto** [comment]
I'm not sure yet.  There is a pattern where we have some tests that we want to make sure work the same way for all packages. But perhaps the right thing to do is have them all distinct. I don't have the willpower to think hard about that tonight. I just want to get each of the packages mostly separated, so it is clear where they go.

**Pull request merged**